### PR TITLE
AG-9072 - Add docs left hand nav

### PR DESCRIPTION
**Pull request by taktran**
## Changes

* Add docs left hand nav
* Add Astro validation for menu data contents
* Add placeholder doc pages for missing content

## Notes

Just the first pass of side menu. In later passes, we'll work on

* Mobile view
* Filtering out javascript only pages

### CODE REVIEW

File: packages/ag-charts-website/src/content/menu/data.json [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+        ]
+    },
+    "charts": {
+        "items": [
+            {
+                "title": "Line",
+                "icon": "chartsLine",
+                "items": [
+                    {
+                        "title": "Overview",
```
Comment (alantreadway) [inline packages/ag-charts-website/src/content/menu/data.json]: `"title": "Overview"` is repeated across all 2nd-level items, why is that?

File: packages/ag-charts-website/src/content/menu/data.json [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+                ]
+            },
+            {
+                "title": "AG Grid Integrated Charts",
+                "url": "https://ag-grid.com/javascript-data-grid/integrated-charts/",
+                "isEnterprise": true
+            }
+        ]
+    },
+    "charts": {
```
Comment (alantreadway) [inline packages/ag-charts-website/src/content/menu/data.json:93]: Is this for the gallery or showcase? (It's not clear to me from the name)

File: packages/ag-charts-website/src/content/menu/data.json (hunk 1 above)
Comment (taktran) [reply packages/ag-charts-website/src/content/menu/data.json]: They are placeholders for now, and may be generated in future. It's easier to change as items in the menu data json for now

File: packages/ag-charts-website/src/content/menu/data.json (hunk 2 above)
Comment (taktran) [reply packages/ag-charts-website/src/content/menu/data.json:93]: Neither. It's just for the left hand nav in docs as it's in `menu/data.json`, but it's just for the charts part of the menu (ie, with icons), which is separate from the main part.

**Review by alantreadway — APPROVED** [review]

**Pull request merged**